addons/binding/org.openhab.binding.icloud/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,94 @@
+	<bridge-type id="account">
+		<label>iCloud Account</label>
+		<description>Provides various information for iCloud devices registered with this account.</description>
+
+		<config-description>
+			<parameter name="appleId" type="text" required="true">
+				<context>email</context>
+				<label>Apple Id</label>
+				<description>Apple Id (e-mail) to access iCloud information.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<context>password</context>
+				<label>Password</label>
+				<description>Apple iCloud password for the given Id.</description>
+			</parameter>
+			<parameter name="refreshTimeInMinutes" type="integer" min="5" max="65535" unit="min">
+				<label>Refresh</label>
+				<description>Refresh time in minutes.</description>
+				<default>5</default>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>iCloud Device</label>
+
+		<channels>
+			<channel typeId="BatteryStatus" id="batteryStatus" />
+			<channel typeId="system.battery-level" id="batteryLevel" />
+			<channel typeId="FindMyPhone" id="findMyPhone" />
+			<channel typeId="Location" id="location" />
+			<channel typeId="LocationAccuracy" id="locationAccuracy" />
+			<channel typeId="LocationLastUpdate" id="locationLastUpdate" />
+			<channel typeId="DeviceName" id="deviceName" />
+		</channels>
+
+		<representation-property>deviceId</representation-property>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device id</label>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+	
+	<channel-type id="Refresh">
+		<item-type>Switch</item-type>
+		<label>Refresh</label>
+		<description>Update all devices registered with this account immediately.</description>
+	</channel-type>
+	<channel-type id="BatteryStatus">
+		<item-type>String</item-type>
+		<label>Battery Status</label>
+	    <state>
+	        <options>
+	            <option value="NotCharging">Not charging</option>
+	            <option value="Charged">Charged</option>
+	            <option value="Charging">Charging</option>
+	            <option value="Unknown">Unknown</option>
+	        </options>
+	    </state>
+	</channel-type>
+	<channel-type id="FindMyPhone">
+		<item-type>Switch</item-type>
+		<label>Find my phone</label>
+	</channel-type>
+	<channel-type id="Location">
+		<item-type>Location</item-type>
+		<label>Location</label>
+	</channel-type>
+	<channel-type id="LocationAccuracy" advanced="true">
+		<item-type>Number</item-type>
+		<label>Location accuracy</label>
+		<state readOnly="true" pattern="%d m" />
+	</channel-type>
+	<channel-type id="LocationLastUpdate" advanced="true">
+		<item-type>DateTime</item-type>
+		<label>Last location update</label>
+	</channel-type>
+	<channel-type id="DeviceName">

```

Comment:
Can the device name be changed? Do users do that regularly? Otherwise, I do not see much sense in this channel.

Reply:
I was the one asking for the information a while back. Back then I was unsure which out of x devices I was seeing data from. Many things happened since then. It would probably be better suited as a thing property.

Reply:
agreed.

Reply:
Done :-)